File: src/main/java/guru/xgm/angelscript/psi/AngelScriptFile.java
package guru.xgm.angelscript.psi;

import com.intellij.extapi.psi.PsiFileBase;
import com.intellij.openapi.fileTypes.FileType;
import com.intellij.psi.FileViewProvider;
import guru.xgm.angelscript.AngelScriptFileType;
import guru.xgm.angelscript.AngelScriptLanguage;
import org.jetbrains.annotations.NotNull;

public class AngelScriptFile extends PsiFileBase {

    public AngelScriptFile(@NotNull FileViewProvider viewProvider) {
        super(viewProvider, AngelScriptLanguage.INSTANCE);
    }

    @NotNull
    @Override
    public FileType getFileType() {
        return AngelScriptFileType.INSTANCE;
    }

    @Override
    public String toString() {
        return "AngelScript File";
    }

}

File: src/main/kotlin/raft/war/language/angelscript/psi/file/AngelScriptFile.kt
package raft.war.language.angelscript.psi.file

import com.intellij.extapi.psi.PsiFileBase
import com.intellij.openapi.fileTypes.FileType
import com.intellij.psi.FileViewProvider
import raft.war.language.angelscript.AngelScriptLanguage
import raft.war.language.angelscript.AngelScriptFileType

class AngelScriptFile(viewProvider: FileViewProvider) : PsiFileBase(viewProvider, AngelScriptLanguage.instance) {
    override fun getFileType(): FileType {
        return AngelScriptFileType.instance
    }

    override fun toString(): String {
        return AngelScriptLanguage.NAME
    }
}
